src/simplification/SimplifyItems.ts

```diff
@@ -12,50 +13,66 @@ type Output = CommonModel | undefined;
-    if (seenSchemas.has(schema)) return seenSchemas.get(schema)!;
+    if (seenSchemas.has(schema)) return seenSchemas.get(schema);
+    const tempOutput: Output = new CommonModel();
     seenSchemas.set(schema, tempOutput);
-    const mergeWithItem = (model : CommonModel) => {
-      tempOutput = CommonModel.mergeCommonModels(tempOutput, model, schema);
-    };
-    const addToItems = (out: Output) => {
-      if (out !== undefined) {
-        mergeWithItem(out);
-      }
-    };
-    const handleCombinationSchemas = (schemas: (Schema | boolean)[] = []) => {
-      schemas.forEach((itemSchema) => {
-        addToItems(simplifyItems(itemSchema, simplifier, seenSchemas));
-      });
-    };
 
+    //Check if any items have been sat, if so lets merge them
     if (schema.items !== undefined) {
-      const addItemsAndModels = (newModels : CommonModel[]) => {
-        mergeWithItem(newModels[0]);
-      };
       if (Array.isArray(schema.items)) {
         schema.items.forEach((value) => {
-          addItemsAndModels(simplifier.simplify(value));
+          mergeWithCurrentModel(simplifier.simplify(value)[0], schema, tempOutput);
         });
       } else {
-        addItemsAndModels(simplifier.simplify(schema.items));
+        mergeWithCurrentModel(simplifier.simplify(schema.items)[0], schema, tempOutput);
       }
     }
 
-    //If we encounter combination schemas ensure we recursively find the properties
-    handleCombinationSchemas(schema.allOf);
-    handleCombinationSchemas(schema.oneOf);
-    handleCombinationSchemas(schema.anyOf);
+    //If we encounter combination schemas ensure we recursively find the items
+    combineSchemas(schema.allOf, tempOutput, simplifier, seenSchemas);
+    combineSchemas(schema.oneOf, tempOutput, simplifier, seenSchemas);
+    combineSchemas(schema.anyOf, tempOutput, simplifier, seenSchemas);
 
-    //If we encounter combination schemas ensure we recursively find the properties
-    if (schema.then) {
-      addToItems(simplifyItems(schema.then, simplifier, seenSchemas));
-    }
-    if (schema.else) {
-      addToItems(simplifyItems(schema.else, simplifier, seenSchemas));
+    //If we encounter conditional schemas ensure we recursively find the items
+    combineSchemas(schema.then, tempOutput, simplifier, seenSchemas);
+    combineSchemas(schema.else, tempOutput, simplifier, seenSchemas);
+    
+    return !Object.keys(tempOutput).length ? undefined : tempOutput;
+  }
+  return undefined;
+}
+
+/**
+ * Go through schema(s) and combine the simplified items together.
+ * 
+ * @param schema to go through
+ * @param currentModel the current output
+ * @param simplifier the simplifier to use
+ * @param seenSchemas schemas which we already have outputs for
+ */
+function combineSchemas(schema: (Schema | boolean) | (Schema | boolean)[] | undefined, currentModel: Output, simplifier : Simplifier, seenSchemas: Map<any, Output>) {
+  if (schema === undefined) return;
+  if (typeof schema === 'boolean') return;
```

Comment:
```suggestion
  if (typeof schema !== 'object') return;
```

Reply:
`schema` can be undefined :smile:

Reply:
but then you have also false for undefined :P

Reply:
AHH, magic 🔥